### PR TITLE
DRILL-7486: Refactor row set reader builders

### PR DESCRIPTION
**Pull request by paul-rogers**
Moves reader building code into a shared location, independent of the RowSet class. Allows readers to be built from a VectorContainer in addition to a row set.

Tests: Reran all unit tests, including those that use the row set readers.

This PR can be reviewed after the 1.17 release is complete.

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/resultSet/model/hyper/HyperReaderBuilder.java [hunk 1]
```diff
@@ -71,30 +72,26 @@
...
-    public int hyperVectorIndex( ) {
-      return AccessorUtilities.sv4Batch(sv4.get(position));
+  public static RowSetReaderImpl build(BatchAccessor batch) {
+    VectorContainer container = batch.container();
+    try {
+      return build(container,
+          new HyperSchemaInference().infer(container),
+          batch.selectionVector4());
+    } catch (SchemaChangeException e) {
+      throw new UnsupportedOperationException(e);
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/main/java/org/apache/drill/exec/physical/resultSet/model/hyper/HyperReaderBuilder.java]: Why is we capture schema change exception, unsupported exception is thrown?

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/rowSet/HyperRowIndex.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  @Override
+  public int offset() {
+    return AccessorUtilities.sv4Index(sv4.get(position));
+  }
+
+  @Override
+  public int hyperVectorIndex( ) {
+    return AccessorUtilities.sv4Batch(sv4.get(position));
+  }
+}
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/main/java/org/apache/drill/exec/physical/rowSet/HyperRowIndex.java]: New line in the end of file

**Comment by arina-ielchiieva** [comment]
@paul-rogers is you address code review comments today, we can include this PR into the release.
cc @vvysotskyi

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/resultSet/model/hyper/HyperReaderBuilder.java (hunk 1 above)
Comment (paul-rogers) [reply exec/java-exec/src/main/java/org/apache/drill/exec/physical/resultSet/model/hyper/HyperReaderBuilder.java]: Revised and added comment.

**Comment by paul-rogers** [comment]
@arina-ielchiieva, addressed the comments. Should be ready to go. Thanks for including this one.

**Comment by arina-ielchiieva** [comment]
+1, LGTM

**Pull request closed without merging**